src/components/vm/vmActions.tsx

```diff
@@ -44,164 +44,134 @@ import { RenameDialog } from './vmRenameDialog.jsx';
-        store.dispatch(
-            updateVm({
-                connectionName: vm.connectionName,
-                name: vm.name,
-                error: {
-                    text: cockpit.format(_("VM $0 failed to start"), vm.name),
-                    detail: ex.message,
-                }
-            })
-        );
-    })
-);
+const domainCanRun = (vmState: VMState, hasInstallPhase: boolean) => !hasInstallPhase && vmState == 'shut off';
+const domainCanShutdown = (vmState: VMState) => domainCanReset(vmState);
+const domainCanPause = (vmState: VMState) => vmState == 'running';
+const domainCanRename = (vmState: VMState) => vmState == 'shut off';
+const domainCanResume = (vmState: VMState) => vmState == 'paused';
+
+function startOperationProgress(vm: VM) {
+    store.dispatch(
+        updateVm({
+            connectionName: vm.connectionName,
+            name: vm.name,
+            operationInProgressFromState: vm.state,
+            onShutOff: null,
+        })
+    );
+}
+
+function isOperationInProgress(vm: VM) {
+    return vm.state == vm.operationInProgressFromState;
+}
+
+function setVmError(vm: VM, msg: string, ex: unknown) {
+    console.warn(msg, ":", String(ex));
+    store.dispatch(
+        updateVm({
+            connectionName: vm.connectionName,
+            name: vm.name,
+            operationInProgressFromState: undefined,
+            onShutOff: null,
+            error: {
+                text: msg,
+                detail: String(ex),
+            }
+        })
+    );
+}
+
+export async function vmStart(vm: VM) {
+    startOperationProgress(vm);
+    try {
+        await vmDomainMethod<void>(vm, 'Create', 'u', 0);
+    } catch (ex) {
+        setVmError(vm, cockpit.format(_("VM $0 failed to start"), vm.name), ex);
+    }
+}
 
-const onInstall = (vm: VM) => (
-    domainInstall({ vm }).catch(ex => {
+async function onInstall(vm: VM) {
+    try {
+        await domainInstall({ vm });
+    } catch (ex) {
         addNotification({
             text: cockpit.format(_("VM $0 failed to get installed"), vm.name),
-            detail: ex.message.split(/Traceback(.+)/)[0],
+            detail: String(ex).split(/Traceback(.+)/)[0],
             resourceId: vm.id,
         });
-    })
-);
+    }
+}
 
-const onReboot = (vm: VM) => (
-    domainReboot({ id: vm.id, connectionName: vm.connectionName }).catch(ex => {
-        store.dispatch(
-            updateVm({
-                connectionName: vm.connectionName,
-                name: vm.name,
-                error: {
-                    text: cockpit.format(_("VM $0 failed to reboot"), vm.name),
-                    detail: ex.message,
-                }
-            })
-        );
-    })
-);
+export async function vmReboot(vm: VM) {
+    try {
+        await vmDomainMethod<void>(vm, 'Reboot', 'u', 0);
+    } catch (ex) {
+        setVmError(vm, cockpit.format(_("VM $0 failed to reboot"), vm.name), ex);
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2376-5e904223-20251103-111358-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)